src/execute_linter.cc

```diff
@@ -24,24 +25,172 @@
+  LinterResult result;
+  for (int i = 0; i < static_cast<int>(sql.size()); ++i) {
+    if (sql[i] == '\'' || sql[i] == '"') {
+      // The sql is inside string. This will ignore sql part until
+      // the same type(' or ") of character will occur
+      // without \ in the beginning. For example 'a"b' is a valid string.
+      for (int j = i + 1; j < static_cast<int>(sql.size()); ++j) {
+        if (sql[j - 1] == '\\' && sql[j] == sql[i]) continue;
+        if (sql[j] == sql[i] || j + 1 == static_cast<int>(sql.size())) {
+          i = j;
+          break;
+        }
+      }
+      continue;
+    }
+
+    // Ignore multiline comments.
+    if (i > 0 && sql[i - 1] == '/' && sql[i] == '*') {
+      // It will start checking after '/*' and after the iteration
+      // finished, the pointer 'i' will be just after '*/' (incrementation
+      // from the for statement is included).
+      i += 2;
+      while (i < static_cast<int>(sql.size()) &&
+             !(sql[i - 1] == '*' && sql[i] == '/')) {
+        ++i;
+      }
+    }
+
+    absl::string_view type = "";
+    if (i > 0 && sql[i - 1] == '-' && sql[i] == '-') type = "--";
+    if (i > 0 && sql[i - 1] == '/' && sql[i] == '/') type = "//";
+    if (sql[i] == '#') type = "#";
+
+    if (type != "") {
+      ++i;
+      while (i < static_cast<int>(sql.size()) && sql[i] == ' ') ++i;
+
+      if (sql.substr(i, 6) == "NOLINT" || sql.substr(i, 4) == "LINT") {
+        bool disabler = (sql.substr(i, 6) == "NOLINT");
+        while (i < static_cast<int>(sql.size()) && sql[i] != '(' &&
+               sql[i] != options.LineDelimeter())
+          ++i;
+        ++i;
+        std::string s = "";
+        while (i < static_cast<int>(sql.size()) && sql[i - 1] != ')' &&
+               sql[i] != options.LineDelimeter()) {
+          if (sql[i] == ' ') continue;
+          if (sql[i] == ',' || sql[i] == ')') {
+            std::map<std::string, ErrorCode> error_map = GetErrorMap();
+            if (!error_map.count(s)) {
+              result.Add(ErrorCode::kNoLint, sql, i,
+                         absl::StrCat("Unkown NOLINT error category: ", s));
+            } else {
+              const ErrorCode& code = error_map[s];
+              if (!nolint->count(code)) (*nolint)[code] = options;
+
+              (*nolint)[code].IsActive(i + 1);
+              if (disabler)
+                (*nolint)[code].Disable(i);
+              else
+                (*nolint)[code].Enable(i);
+              (*nolint)[code].IsActive(i + 1);
+            }
+            s = "";
+          } else {
+            s += sql[i];
+          }
+          ++i;
+        }
+      }
+      while (i < static_cast<int>(sql.size()) &&
+             sql[i] != options.LineDelimeter())
+        ++i;
+      continue;
+    }
+  }
+
+  return result;
+}
+
+LinterOptions GetOptionsFromConfig() {
+  // TODO(orhanuysal): Connect here with a config file.
+  return LinterOptions();
+}
+
+// TODO(orhanuysal): Move these classes to proper .h file
```

Comment:
well, check_list.h ? in general, the filnename follows the name of the main type declared there.

Reply:
I actually feared this PR would seem huge, since each file automaticly creates at least 20-30 lines. I will make it into check_list.h and check_list.cc

Reply:
In re-naming PR :) not in this

Reply:
pro-tip for the future: you could have a separate pr that just defines CheckList, but is not using it in the code :)